src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -14,224 +14,199 @@
-               callback.onResponseReceived(spellCheckerResult);
+               spellCheckerResult.getCorrect().add(word);
             }
-            
-            @Override
-            public void onError(ServerError error)
+            else
             {
-               callback.onError(error);
+               spellCheckerResult.getIncorrect().add(word);
             }
-            
-         });
-   }
-   
-   public void suggestionList(String word,
-                              ServerRequestCallback<JsArrayString> callback)
-   {
-      spellingService_.suggestionList(word, callback);
+         }
+      }
+      callback.onResponseReceived(spellCheckerResult);
    }
-   
+
    public void addToUserDictionary(final String word)
    {
-      userDictionary_.append(word);  
+      userDictionary_.append(word);
    }
-   
+
    public void addIgnoredWord(String word)
    {
       contextDictionary_.add(word);
       context_.writeDictionary(contextDictionary_);
-      
+
       updateIgnoredWordsIndex();
 
       context_.invalidateMisspelledWords();
    }
-  
+
+   public String[] suggestionList(String word)
+   {
+      if (typoNative == null)
+      {
+         return new String[0];
+      }
+
+      return typoNative.suggest(word);
+   }
    private boolean isWordIgnored(String word)
    {
-      if (allIgnoredWords_.contains(word))
-         return true;
-      else if (ignoreUppercaseWord(word))
-         return true;
-      else if (ignoreWordWithNumbers(word))
-         return true;
-      else
-         return false;
+      return (allIgnoredWords_.contains(word) ||
+              ignoreUppercaseWord(word) ||
+              ignoreWordWithNumbers(word));
    }
-   
+
    private boolean ignoreUppercaseWord(String word)
    {
       if (!uiPrefs_.ignoreWordsInUppercase().getValue())
          return false;
-     
+
       for (char c: word.toCharArray())
       {
          if(!Character.isUpperCase(c))
             return false;
       }
       return true;
    }
-   
+
    private boolean ignoreWordWithNumbers(String word)
    {
       if (!uiPrefs_.ignoreWordsWithNumbers().getValue())
          return false;
-      
+
       for (char c: word.toCharArray())
       {
          if(Character.isDigit(c))
             return true;
       }
       return false;
    }
-  
+
    private void updateIgnoredWordsIndex()
    {
       allIgnoredWords_.clear();
       allIgnoredWords_.addAll(userDictionaryWords_);
       allIgnoredWords_.addAll(contextDictionary_);
    }
-    
-   private ValueChangeHandler<Boolean> prefChangedHandler_ = 
-                                       new ValueChangeHandler<Boolean>() {
+
+   private ValueChangeHandler<Boolean> prefChangedHandler_ = new ValueChangeHandler<Boolean>() {
       @Override
       public void onValueChange(ValueChangeEvent<Boolean> event)
       {
          context_.invalidateAllWords();
-      }  
+      }
    };
 
    private final Context context_;
-   
+
+   private TypoNative typoNative;
```

Comment:
nit: member variable should have trailing `_`

Reply:
👍